quinn-proto/src/connection.rs

```diff
@@ -376,13 +383,22 @@ impl Connection {
     }
 
     /// Initiate a connection
-    pub fn connect(&mut self, ctx: &Context, server_name: &str) -> Result<(), ConnectError> {
-        let mut tls = TlsSession::new_client(
-            &ctx.config.tls_client_config,
-            server_name,
-            &TransportParameters::new(&ctx.config),
-        ).unwrap();
+    pub fn connect(
+        &mut self,
+        ctx: &Context,
+        config: &Arc<ClientConfig>,
+        server_name: &str,
+    ) -> Result<(), ConnectError> {
+        assert_eq!(
```

Comment:
If we're asserting this here, should we have a similar assert for the server side?

Reply:
In fact, rustls contains a similar assert already, so perhaps this should be removed.